README.md

```diff
@@ -1,106 +1,131 @@
-# Superformula Back-end Developer Test
+# Superformula Cloud-native User API
 
-Be sure to read **all** of this document carefully, and follow the guidelines within.
+A RESTful API that can `create/read/update/delete` user data from a persistence store.
 
-## Context
+## Approach
 
-Build a RESTful API that can `create/read/update/delete` user data from a persistence store.
+**CloudFormation**
 
-### User Model
+* All resource management is done via a CloudFormation template: [superformula-api.template](templates/superformula-api.template).
+* There are three optional features of the API that are enabled via parameters:
+    * API key authentication - When enabled, resources are created to control authentication via ApiGateway API keys.
+    * Alarms - When enabled, an SNS topic and 4 CloudWatch alarms are created.
+    * DynamoDB auto-scaling - When enabled, the database will be configured to auto-scale. 
 
-```
-{
-  "id": "xxx",                  // user ID (must be unique)
-  "name": "backend test",       // user name
-  "dob": "",                    // date of birth
-  "address": "",                // user address
-  "description": "",            // user description
-  "createdAt": ""               // user created date
-  "updatedAt": ""               // user updated date
-}
-```
+**CloudFront**  
 
-### Functionality
+* Both the API and documentation site are delivered through the same CloudFront distribution.
+* The API _requires_ an https connection, where as the documentation site will _redirect_ the user to https automatically.
+* All _GET_ requests are cached. This includes both API and documentation requests.
+* The distribution will forward any querystring parameters onto the API (for _GET_ filtering).
 
-- The API should follow typical RESTful API design pattern.
-- The data should be saved in the DB.
-- Provide proper API documentation.
-- Proper error handling should be used.
+**ApiGateway**
```

Comment:
If this project needed to pivot away from REST and be a Graph API instead, how would you approach that?

For example: 
What tools/services would you use? 
Which aspects of your code would need to entirely change? 
Which aspects of your code could remain or be easily adapted?

Reply:
Unfortunately I don't have a great answer for this as I don't have experience beyond some light reading about Graph APIs or GraphQL. However, according to the [Serverless blog](https://serverless.com/blog/make-serverless-graphql-api-using-lambda-dynamodb/), I can continue using API Gateway and DynamoDB. It looks like what changes the most is the models and lambda functions. I think I can define GraphQL Schemas and Mutations along side the current schema definitions and go from there.